File: 977_SquaresOfSortedArray/squaresOfSortedArray.js
var sortedSquares = function(nums) {
    const n = nums.length
    let left = 0
    let right = n - 1
    let res = new Array(n).fill(0)

    for(let i = n - 1; i >= 0; i--) {
        let square
        if (Math.abs(nums[left]) < Math.abs(nums[right])) {
            square = nums[right]
            right--
        } else {
            square = nums[left]
            left++
        }
        res[i] = square * square
    }
    return res
};

File: 977_SquaresOfSortedArray/squaresOfSortedArray.py
def sortedSquares(nums: List[int]) -> List[int]:
    n = len(nums)
    left = 0
    right = n - 1
    res = [0] * n

    for i in range(n - 1, -1, -1):
        if abs(nums[left]) < abs(nums[right]):
            square = nums[right]
            right -= 1
        else:
            square = nums[left]
            left += 1
        res[i] = square * square
    
    return res

print(sortedSquares([-4,-1,0,3,10]))
print(sortedSquares([-7,-3,2,3,11]))